overlays/nodejs/lambda/lambdaMixins.ts

```diff
@@ -0,0 +1,365 @@
+     * passed in JavaScript callback.
+     */
+    code?: never;
+
+    /**
+     * Not allowed when creating an aws.serverless.Function.  The [code] will be generated from the
+     * passed in JavaScript callback.
+     */
+    handler?: never;
+
+    /**
+     * The Javascript function instance to use as the entrypoint for the AWS Lambda out of.  Either
+     * [entryPoint] or [entryPointFactory] must be provided.
+     */
+    entryPoint?: EntryPoint<E, R>;
+
+    /**
+     * The Javascript function instance that will be called to produce the function that is the
+     * entrypoint for the AWS Lambda. Either [entryPoint] or [entryPointFactory] must be provided.
+     *
+     * This form is useful when there is expensive initialization work that should only be executed
+     * once.  The factory-function will be invoked once when the final AWS Lambda module is loaded.
+     * It can run whatever code it needs, and will end by returning the actual function that Lambda
+     * will call into each time the Lambda is invoked.
+     */
+    entryPointFactory?: EntryPointFactory<E, R>;
+
+    /**
+     * A pre-created role to use for the Function. If not provided, [policies] will be used.
+     */
+    role?: iam.Role;
+
+    /**
+     * A list of IAM policy ARNs to attach to the Function.  Will be used if [role] is not provide.
+     * If neither [role] nor [policies] is provided, a default policy of [iam.AWSLambdaFullAccess]
+     * will be used instead.
+     */
+    policies?: arn.ARN[];
+
+    /**
+     * The Lambda runtime to use.  If not provided, will default to [NodeJS8d10Runtime]
+     */
+    runtime?: runtime.Runtime;
+
+    /**
+     * Options to control which paths/packages should be included or excluded in the zip file containing
+     * the code for the AWS lambda.
+     */
+    codePathOptions?: pulumi.runtime.CodePathOptions;
+}>;
+
+/**
+ * Base type for all subscription types.  An event subscription represents a connection between some
+ * AWS resource an an AWS lambda that will be triggered when something happens to that resource.
+ */
+export class EventSubscription extends pulumi.ComponentResource {
+    public permission: permission.Permission;
+    public func: LambdaFunction;
+
+    public constructor(
+        type: string, name: string, props: Record<string, any>, opts?: pulumi.ResourceOptions) {
+
+        super(type, name, props, opts);
+    }
+}
+
+/* @internal */ export function createFunctionFromEventHandler<E, R>(
+    name: string, handler: EventHandler<E, R>, opts?: pulumi.ResourceOptions): LambdaFunction {
+
+    if (handler instanceof Function) {
+        return createFunction(name, { entryPoint: handler } , opts);
+    }
+    else {
+        return handler;
+    }
+}
+
+/**
+ * Creates an AWS Lambda out of the provider options.func or options.factoryFunc.  Configuration of
+ * this lambda can be provided through [options].
+ */
+export function createFunction<E, R>(
```

Comment:
Along the lines of other comments, I do wonder whether this should use some other identifying word - like `createCallbackFunction`?  It's argument bag doesn't allow for creating a Function from normal handler/code data - so it's really disjoint from `new aws.lambda.Function`, not something that encapsulates that plus additional use cases.

Frankly, it's almost like this should just be `new aws.lambda.CallbackFunction()` for full consistency with our normal pattern?  With the design of things as they are in this library - what's the main argument against that?

Reply:
> Frankly, it's almost like this should just be new aws.lambda.CallbackFunction() for full consistency with our normal pattern? With the design of things as they are in this library - what's the main argument against that?

I guess one question is what does it return - but a subclass of `aws.lambda.Function` seems a natural answer - right?

Reply:
That would be an interesting approach.  I like it, let me go try that out!